inst/templates/rosepine.R

```diff
@@ -0,0 +1,500 @@
+#' in August 2024.
+#' @source https://rosepinetheme.com/
+#'
+rose_pine <- list(
+  base = "#191724", # main panels, frames, sidebars, tabs
+  surface = "#1f1d2e", # panels not related to focal context
+  overlay = "#26233a", # panels temporary in nature (e.g. notifications/dialogs)
+  muted = "#6e6a86", # comments, ignored content
+  subtle = "#908caa", # secondary content: operators, punctuation, tab names
+  text = "#e0def4", # primary content: normal text, variables
+  love = "#eb6f92", # term red, diagnostic errors, builtins
+  gold = "#f6c177", # term yellow, diagnostic warnings, strings
+  rose = "#ebbcba", # term cyan, matching search background, booleans, functions
+  pine = "#31748f", # term green, conditionals, keywords
+  foam = "#9ccfd8", # term blue, diagnostic info, keys, tags, types
+  iris = "#c4a7e7", # term magenta, links, hints, methods, params
+  highlight_low = "#21202e", # cursorline background
+  highlight_med = "#403d52", # selection background paired with text foreground
+  highlight_high = "#524f67" # borders/dividers, cursor background
+)
+
+rose_pine_moon <- list(
+  base = "#232136",
+  surface = "#2a273f",
+  overlay = "#393552",
+  muted = "#6e6a86",
+  subtle = "#908caa",
+  text = "#e0def4",
+  love = "#eb6f92",
+  gold = "#f6c177",
+  rose = "#ea9a97",
+  pine = "#3e8fb0",
+  foam = "#9ccfd8",
+  iris = "#c4a7e7",
+  highlight_low = "#2a283e",
+  highlight_med = "#44415a",
+  highlight_high = "#56526e"
+)
+
+rose_pine_dawn <- list(
+  base = "#faf4ed",
+  surface = "#fffaf3",
+  overlay = "#f2e9e1",
+  muted = "#9893a5",
+  subtle = "#797593",
+  text = "#575279",
+  love = "#b4637a",
+  gold = "#ea9d34",
+  rose = "#d7827e",
+  pine = "#286983",
+  foam = "#56949f",
+  iris = "#907aa9",
+  highlight_low = "#f4ede8",
+  highlight_med = "#dfdad9",
+  highlight_high = "#cecacd"
+)
+
+#' Create Rosé Pine .rstheme file
+#'
+#' @param variant A string - A string defining the theme palette variant to use;
+#' must be "base", "moon", or "dawn";
+#' see [Rosé Pine Variants](https://rosepinetheme.com/palette/)
+#' @param apply A Boolean - generate and install/apply the .rstheme file (TRUE),
+#' or just save the .rstheme file without applying (FALSE); defaults to FALSE
+#' @param as_sass A Boolean - save theme as .sass file and do not apply (TRUE);
+#' defaults to FALSE
+#' @param ... Additional arguments to pass to [rsthemes::rstheme()] overriding
+#' defaults if necessary
+#'
+#' @export
+rosepine_rstheme <- function(
+    variant = "base",
+    apply = FALSE,
+    as_sass = FALSE,
+    ...) {
+  variant.choices <- list("base" = NULL, "moon" = "Moon", "dawn" = "Dawn")
+  stopifnot(
+    "Variant must be 'base', 'moon', or 'dawn'." =
+      tolower(variant) %in% names(variant.choices)
+  )
+  variant <- do.call(switch, c(tolower(variant), variant.choices))
+
+  theme_dark <- !identical(variant, "Dawn")
+
+  rp_pal <- get(paste0(
+    "rose_pine",
+    if (!is.null(variant)) "_",
+    tolower(variant)
+  ))
```

Comment:
Similarly, this code is less clever but easier to read and maintain.

```suggestion
  rp_pal <- switch(
    variant,
    Dawn = rose_pine_dawn,
    Moon = rose_pine_moon,
    rose_pine
  )
```